syntax/arrai.wbnf

```diff
@@ -10,7 +10,7 @@ expr   -> C* amp="&"* @ C* arrow=(
         > C* @:binop="&&" C*
         > C* @:compare=/{!?(?:<:|=|<=?|>=?|\((?:<=?|>=?|<>=?)\))} C*
         > C* @ if=("if" t=expr ("else" f=expr)?)* C*
-        > C* @ cond=("cond" "(" (key=(("(" expr:",", ")") | expr) ":" value=expr):",",? ("*" ":" f=expr ","?)? ")")? C*
+        > C* @ cond=("cond" "{" (condition=(("(" expr:",", ")") | "("expr")" | pattern) ":" value=expr):",",? ("_" ":" f=expr ","?)? "}")? C*
```

Comment:
`"(" expr ")"` should be part of `patternterms`. `identpattern=IDENT` should be something like `exprpattern=expr`. We should tidy this up separately from this PR and then leverage it.